lib/dalli/protocol/value_serializer.rb

```diff
@@ -33,42 +33,17 @@ def store(value, req_options, bitflags)
         [store_value, bitflags]
       end
 
-      # TODO: Some of these error messages need to be validated.  It's not obvious
-      # that all of them are actually generated by the invoked code
-      # in current systems
-      # rubocop:disable Layout/LineLength
-      TYPE_ERR_REGEXP = %r{needs to have method `_load'|exception class/object expected|instance of IO needed|incompatible marshal file format}.freeze
-      ARGUMENT_ERR_REGEXP = /undefined class|marshal data too short/.freeze
-      NAME_ERR_STR = 'uninitialized constant'
-      # rubocop:enable Layout/LineLength
-
       def retrieve(value, bitflags)
         serialized = (bitflags & FLAG_SERIALIZED) != 0
-        serialized ? serializer.load(value) : value
-      rescue TypeError => e
-        filter_type_error(e)
-      rescue ArgumentError => e
-        filter_argument_error(e)
-      rescue NameError => e
-        filter_name_error(e)
-      end
-
-      def filter_type_error(err)
-        raise err unless TYPE_ERR_REGEXP.match?(err.message)
-
-        raise UnmarshalError, "Unable to unmarshal value: #{err.message}"
-      end
-
-      def filter_argument_error(err)
-        raise err unless ARGUMENT_ERR_REGEXP.match?(err.message)
-
-        raise UnmarshalError, "Unable to unmarshal value: #{err.message}"
-      end
-
-      def filter_name_error(err)
-        raise err unless err.message.include?(NAME_ERR_STR)
-
-        raise UnmarshalError, "Unable to unmarshal value: #{err.message}"
+        if serialized
+          begin
+            serializer.load(value)
+          rescue StandardError => e
+            raise UnmarshalError, "Unable to unmarshal value: #{e.message}"
```

Comment:
```suggestion
          rescue => error
            raise UnmarshalError, "Unable to unmarshal value!"
```

Cause will already carry the message, no point repeating it.

Reply:
(The `StandardError` mention is a nod to rubocop rules.)

Reply:
Actually we don't even need `=> error` if we are just raising a more general error :)

Reply:
🧠 == 🧠 (I pushed new.)